metaseq/tasks/streaming_language_modeling.py

```diff
@@ -334,6 +337,7 @@ def load_dataset(self, split: str, epoch=1, combine=False, **kwargs):
 
         dataset = torch.utils.data.ConcatDataset(datasets)
 
+        dataset = DeferredDataset(dataset)
```

Comment:
let's leave a few comments explaining the relationship between deferred / skip differed datasets and why we're employing them